Cargo.toml

```diff
@@ -13,7 +13,18 @@ lalrpop-util = {version = "^0.19", features = ["lexer"]}
 lazy_static = "^1.4.0"
 mktemp = "0.5.0"
 num-bigint = "^0.4"
+polyexen = { git = "https://github.com/Dhole/polyexen", rev="c4864d2debf9ae21138c9d67a99c93c3362df19b"}
+halo2_proofs = "0.2"
+prettytable = "0.10.0"
+num-traits = "0.2.15"
+num-integer = "0.1.45"
 regex = "^1.7.0"
 
 [build-dependencies]
 lalrpop = "^0.19"
+
+[patch."https://github.com/privacy-scaling-explorations/halo2.git"]
+halo2_proofs = { git ="https://github.com/ed255/halo2", rev = "63e969673de83a410f21553fabec8f4b35bda1a5" }
+
+[patch.crates-io]
+halo2_proofs = { git ="https://github.com/ed255/halo2", rev = "63e969673de83a410f21553fabec8f4b35bda1a5" }
```

Comment:
@adria0 is there a reason for the specific rev? Why not just use the latest version of the repo?

Reply:
My fork of halo2 contains 2 sets of changes:
- 1. Make some fields in structs public via getters.  This should be easily upstreamable and it's a non-breaking change for halo2
- 2. Change the trait `Circuit` so that the `configure` method takes `&self`.  This is a breaking change.  This is required In order to have halo2 circuits that are defined at runtime.  Otherwise the `configure` method can only be parametrized with associated const of the type.  There are other options here, but I think adding `&self` is the most straightforward.  This may take some time to be upstreamed in PSE halo2 (IMO it requires debating which is the best option, as there are alternatives).